### PR TITLE
Add efs-provisioner to eks-aws module

### PR DESCRIPTION
**Pull request by raphink**

### CODE REVIEW

File: argocd/argocd/values.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ---
 argo-cd:
-  installCRDs: false
+  installCRDs: true
```
Comment (mcanevet) [inline argocd/argocd/values.yaml:3]: I can't remember why I set this to `false` but I'm pretty sure it's required.

**Review by mcanevet — CHANGES_REQUESTED** [review]

File: argocd/argocd/values.yaml (hunk 1 above)
Comment (raphink) [reply argocd/argocd/values.yaml:3]: That was an attempt to fix the stack, as I'm currently getting errors due to the Application CRD not being deployed.

File: modules/eks-aws/efs.tf [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module "efs" {
+  count = var.enable_efs ? 1 : 0
+  source = "./efs"
```
[the remark this answers is not in the document]
Comment (raphink) [reply modules/eks-aws/efs.tf:3]: There's already one by another company, though it looked unnecessarily complicated for our needs.

File: modules/eks-aws/main.tf [hunk 3]
```diff
@@ -107,7 +107,7 @@ resource "helm_release" "argocd" {
   name              = "argocd"
   chart             = "${path.module}/../../argocd/argocd"
   namespace         = "argocd"
-  dependency_update = true
+  #dependency_update = true
```
[the remark this answers is not in the document]
Comment (raphink) [reply modules/eks-aws/main.tf:110]: In an attempt to fix the issues I'm seeing currently...

[the remark this answers is not in the document]
Comment (raphink) [reply modules/eks-aws/main.tf:110]: (see Slack)

File: argocd/argocd/values.yaml (hunk 1 above)
Comment (mcanevet) [reply argocd/argocd/values.yaml:3]: Are you using master?

File: modules/eks-aws/efs.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (mcanevet) [reply modules/eks-aws/efs.tf:3]: My point was not to use one from another company, but push yours to the registry.

File: argocd/argocd/values.yaml (hunk 1 above)
Comment (raphink) [reply argocd/argocd/values.yaml:3]: Yes

File: modules/eks-aws/efs.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (raphink) [reply modules/eks-aws/efs.tf:3]: Yes, I understand. And I can split it from this pr as soon as it's working 😁

**Review by mcanevet — APPROVED** [review]

**Pull request merged**